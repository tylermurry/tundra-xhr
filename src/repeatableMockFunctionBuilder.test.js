import buildRepeatMap from './requestRepeatMapBuilder';
import buildRepeatableMockFunction from './repeatableMockFunctionBuilder';
import buildRequestId from './requestIdBuilder';

const buildRequest = (method, url) => ({ request: { method, url } });

describe('repeatableMockFunctionBuilder', () => {
  const requests = [
    buildRequest('GET', 'http://some.url'),
    buildRequest('GET', 'http://some.url'),
  ];

  const response = {
    statusCode: 200,
    headers: [],
    content: 'something',
  };

  it('should build a function when no input is given', () => {
    const repeatMap = buildRepeatMap(requests);
    const { request } = requests[0];

    expect(`${buildRepeatableMockFunction(null, repeatMap, request, response)}`).toMatchSnapshot();
  });

  it('should build a function when a repeatMode of \'first\' is given', () => {
    const repeatMap = buildRepeatMap(requests);
    const { request } = requests[0];
    const config = { repeatMode: 'FIRST' };

    expect(`${buildRepeatableMockFunction(config, repeatMap, request, response)}`).toMatchSnapshot();
  });

  it('should build a function when a repeatMode of \'last\' is given '
    + 'and invocations do not exceed the number of repeats', () => {
    const repeatMap = buildRepeatMap(requests);
    const { request } = requests[0];
    const config = { repeatMode: 'LAST' };

    expect(`${buildRepeatableMockFunction(config, repeatMap, request, response)}`).toMatchSnapshot();
  });

  it('should build a function when a repeatMode of \'last\' is given '
    + 'and invocations exceed the number of repeats', () => {
    const repeatMap = buildRepeatMap(requests);
    const { request } = requests[0];
    const config = { repeatMode: 'LAST' };

    repeatMap[buildRequestId(request)].invocations = 3;

    expect(`${buildRepeatableMockFunction(config, repeatMap, request, response)}`).toMatchSnapshot();
  });
});
