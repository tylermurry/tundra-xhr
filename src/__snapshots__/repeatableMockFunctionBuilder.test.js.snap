// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`repeatableMockFunctionBuilder should build a function when a repeatMode of 'first' is given 1`] = `
"function baseMockFunction(req, res) {
    /* istanbul ignore next */cov_1e3dtz6vho.f[2]++;
    cov_1e3dtz6vho.s[4]++;
    return buildResponse(res, response);
  }"
`;

exports[`repeatableMockFunctionBuilder should build a function when a repeatMode of 'last' is given and invocations do not exceed the number of repeats 1`] = `
"function (req, res) {
        if (callCount === 0) {
            ++callCount;
            return typeof mock === 'function'
                ? mock(req, res)
                : createResponseFromObject_1.createResponseFromObject(mock);
        }
        else {
            return undefined;
        }
    }"
`;

exports[`repeatableMockFunctionBuilder should build a function when a repeatMode of 'last' is given and invocations exceed the number of repeats 1`] = `
"function baseMockFunction(req, res) {
    /* istanbul ignore next */cov_1e3dtz6vho.f[2]++;
    cov_1e3dtz6vho.s[4]++;
    return buildResponse(res, response);
  }"
`;

exports[`repeatableMockFunctionBuilder should build a function when no input is given 1`] = `
"function (req, res) {
        if (callCount === 0) {
            ++callCount;
            return typeof mock === 'function'
                ? mock(req, res)
                : createResponseFromObject_1.createResponseFromObject(mock);
        }
        else {
            return undefined;
        }
    }"
`;
